core/src/main/java/org/apache/gravitino/storage/relational/mapper/CatalogMetaBaseProvider.java

```diff
@@ -0,0 +1,172 @@
+        + " #{catalogMeta.catalogName},"
+        + " #{catalogMeta.metalakeId},"
+        + " #{catalogMeta.type},"
+        + " #{catalogMeta.provider},"
+        + " #{catalogMeta.catalogComment},"
+        + " #{catalogMeta.properties},"
+        + " #{catalogMeta.auditInfo},"
+        + " #{catalogMeta.currentVersion},"
+        + " #{catalogMeta.lastVersion},"
+        + " #{catalogMeta.deletedAt}"
+        + " )";
+  }
+
+  public String insertCatalogMetaOnDuplicateKeyUpdate(@Param("catalogMeta") CatalogPO catalogPO) {
+    return "INSERT INTO "
+        + TABLE_NAME
+        + "(catalog_id, catalog_name, metalake_id,"
+        + " type, provider, catalog_comment, properties, audit_info,"
+        + " current_version, last_version, deleted_at)"
+        + " VALUES("
+        + " #{catalogMeta.catalogId},"
+        + " #{catalogMeta.catalogName},"
+        + " #{catalogMeta.metalakeId},"
+        + " #{catalogMeta.type},"
+        + " #{catalogMeta.provider},"
+        + " #{catalogMeta.catalogComment},"
+        + " #{catalogMeta.properties},"
+        + " #{catalogMeta.auditInfo},"
+        + " #{catalogMeta.currentVersion},"
+        + " #{catalogMeta.lastVersion},"
+        + " #{catalogMeta.deletedAt}"
+        + " )"
+        + " ON DUPLICATE KEY UPDATE"
+        + " catalog_name = #{catalogMeta.catalogName},"
+        + " metalake_id = #{catalogMeta.metalakeId},"
+        + " type = #{catalogMeta.type},"
+        + " provider = #{catalogMeta.provider},"
+        + " catalog_comment = #{catalogMeta.catalogComment},"
+        + " properties = #{catalogMeta.properties},"
+        + " audit_info = #{catalogMeta.auditInfo},"
+        + " current_version = #{catalogMeta.currentVersion},"
+        + " last_version = #{catalogMeta.lastVersion},"
+        + " deleted_at = #{catalogMeta.deletedAt}";
+  }
+
+  public String updateCatalogMeta(
+      @Param("newCatalogMeta") CatalogPO newCatalogPO,
+      @Param("oldCatalogMeta") CatalogPO oldCatalogPO) {
+    return "UPDATE "
+        + TABLE_NAME
+        + " SET catalog_name = #{newCatalogMeta.catalogName},"
+        + " metalake_id = #{newCatalogMeta.metalakeId},"
+        + " type = #{newCatalogMeta.type},"
+        + " provider = #{newCatalogMeta.provider},"
+        + " catalog_comment = #{newCatalogMeta.catalogComment},"
+        + " properties = #{newCatalogMeta.properties},"
+        + " audit_info = #{newCatalogMeta.auditInfo},"
+        + " current_version = #{newCatalogMeta.currentVersion},"
+        + " last_version = #{newCatalogMeta.lastVersion},"
+        + " deleted_at = #{newCatalogMeta.deletedAt}"
+        + " WHERE catalog_id = #{oldCatalogMeta.catalogId}"
+        + " AND catalog_name = #{oldCatalogMeta.catalogName}"
+        + " AND metalake_id = #{oldCatalogMeta.metalakeId}"
+        + " AND type = #{oldCatalogMeta.type}"
+        + " AND provider = #{oldCatalogMeta.provider}"
+        + " AND catalog_comment = #{oldCatalogMeta.catalogComment}"
```

Comment:
I think you should merge back that fix you made to here.

Reply:
OK, I will rebase the main branch.